ocaml/xapi/xapi_host.ml

```diff
@@ -1126,10 +1126,33 @@ let crl_list ~__context ~host =
   Certificates.(local_list CRL)
 
 let certificate_sync ~__context ~host =
-  Certificates.local_sync()
+  Certificates.local_sync ()
 
 let get_server_certificate ~__context ~host =
-  Certificates.get_server_certificate()
+  Certificates.get_server_certificate ()
+
+let install_server_certificate ~__context ~host ~certificate ~private_key ~certificate_chain =
+  let pem_chain = match certificate_chain with
+  | ""        -> None
+  | pem_chain -> Some pem_chain
+  in
+  let certificate =
+    Certificates.install_server_certificate ~pem_chain certificate private_key
+  in
+  let date_of_ptime time = Date.of_float (Ptime.to_float_s time) in
+  let map f (a, b) = (f a, f b) in
```

Comment:
personally i think `let date_of_ptime (a, b) = (date_of_ptime a, date_of_ptime b)` is better